niceman/interface/login.py

```diff
@@ -72,10 +77,15 @@ def __call__(resource, resource_id=None, config=None):
         # Get configuration and environment inventory
         # TODO: this one would ask for resource type whenever it is not found
         #       why should we???
-        resource_info, inventory = ResourceManager.get_resource_info(config, resource, resource_id)
+        config, inventory = ResourceManager.get_resource_info(config, resource, resource_id)
 
         # Connect to resource environment
-        env_resource = ResourceManager.factory(resource_info)
+        env_resource = ResourceManager.factory(config)
+
+        # Set resource properties to any backend specific command line arguments.
+        if backend:
+            config = backend_set_config(backend, env_resource, config)
```

Comment:
same applies as above... I feel that it all might become also shorter after we look into making ResourceManager  a proper class etc, but that later